# This is run to avoid an edge case where the latest version of a (probably sub)
# dependency being yanked leads to pip failing to backtrack the dep.
# This happened with `cachecontrol==0.12.12 ; python_version >= "3.7" and python_version < "4.0"`.
name: Verify dependency locks

concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true

on:
  pull_request:
    branches:
      - master
    paths: ["dev-requirements/*.txt"]

jobs:
  verify-pr-dep-changes:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9
        with:
          submodules: "true"

      - name: Set up Python 3.9
        uses: actions/setup-python@61a6322f88396a6271a6ee3565807d608ecaddd1
        with:
          python-version: "3.9"

      - name: install prerequisites
        run: |
          python -m pip install --upgrade pip wheel
          python -m pip install -r ./piped/python/base-requirements/nox.txt

      - name: Verify dependency locks
        run: python -m nox -s verify-deps
